x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/eql/eql.ts

```diff
@@ -110,10 +126,36 @@ export const eqlExecutor = async ({
       result.searchAfterTimes = [eqlSearchDuration];
 
       let newSignals: Array<WrappedFieldsLatest<BaseFieldsLatest>> | undefined;
-      if (response.hits.sequences !== undefined) {
-        newSignals = wrapSequences(response.hits.sequences, buildReasonMessageForEqlAlert);
-      } else if (response.hits.events !== undefined) {
-        newSignals = wrapHits(response.hits.events, buildReasonMessageForEqlAlert);
+
+      const { events, sequences } = response.hits;
+
+      if (events) {
+        if (isAlertSuppressionActive) {
+          await bulkCreateSuppressedAlertsInMemory({
+            enrichedEvents: events,
+            toReturn: result,
+            wrapHits,
+            bulkCreate,
+            services,
+            buildReasonMessage: buildReasonMessageForEqlAlert,
+            ruleExecutionLogger,
+            tuple,
+            alertSuppression: completeRule.ruleParams.alertSuppression,
+            wrapSuppressedHits,
+            alertTimestampOverride,
+            alertWithSuppression,
+            experimentalFeatures,
+          });
+        } else {
+          newSignals = wrapHits(events, buildReasonMessageForEqlAlert);
+        }
+      } else if (sequences) {
+        if (isAlertSuppressionActive && completeRule.ruleParams.alertSuppression) {
```

Comment:
I don't think `completeRule.ruleParams.alertSuppression` is needed here.

`isAlertSuppressionActive` is computed before and uses alertSuppression config for this

Reply:
3f731230b4e